forc/src/ops/forc_build.rs

```diff
@@ -57,16 +57,24 @@ pub fn build(command: BuildCommand) -> Result<pkg::Compiled> {
 
     // If necessary, construct a new build plan.
     let plan: pkg::BuildPlan = plan_result.or_else(|e| -> Result<pkg::BuildPlan> {
-        println!("  Creating a new `Forc.lock` file");
-        println!("    Cause: {}", e);
+        if !command.silent_mode {
+            let cause = if e.to_string().contains("No such file or directory") {
+                anyhow!("lock file did not exist")
+            } else {
+                e
+            };
```

Comment:
I think this is fine for now, however there is a whole range of possible errors beyond the "No such file or directory" error so eventually we should remove this special case that only handles lock file creation in favour of a more structured approach for creating a short-form cause message. This is the kind of thing that #1201 would help a lot with.